--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,11 +33,6 @@

 include(CheckLanguage)

-check_language(ASM)
-if(CMAKE_ASM_COMPILER)
-  enable_language(ASM)
-endif()
-
 if(MSVC)
   if(WavPack_CPU_X86 OR WavPack_CPU_X64)
     enable_language(ASM_MASM)
@@ -45,6 +40,11 @@
     enable_language(ASM_MARMASM)
   endif()
 else()
+  check_language(ASM)
+  if(CMAKE_ASM_COMPILER)
+    enable_language(ASM)
+  endif()
+
   check_language(ASM-ATT)
   if(CMAKE_ASM-ATT_COMPILER)
     enable_language(ASM-ATT)
